### PR TITLE
Worksheet changes.

### PR DESCRIPTION
**Pull request by sagarkoshti1990**
## Description

Worksheet changes

### Changes

 - Teaching
 - Teaching Detail
 - Create Worksheet
 - Worksheet Question Bank
 - Question Bank
 - Worksheet

### CODE REVIEW

File: packages/common-lib/src/components/FilterButton.js [hunk 1]
```diff
@@ -35,11 +46,17 @@ export default function FilterButton({
             colorScheme='button'
             variant='outline'
             px='5'
+            _text={{
```
Comment (ChakshuGautam) [inline packages/common-lib/src/components/FilterButton.js:49]: Possible to use the same `H1`, `H2` that we created earlier here.

File: packages/common-lib/src/components/FilterButton.js [hunk 2]
```diff
@@ -35,11 +46,17 @@ export default function FilterButton({
...
+              fontWeight: '500',
+              fontSize: '14px'
+            }}
             onPress={(e) => setFiltered(true)}
             rightIcon={<IconByName name='ArrowDownSLineIcon' isDisabled />}
             {..._button}
+            {..._filterButton}
           >
-            {t('FILTER')}
+            {filterButtonText ? filterButtonText : t('FILTER')}
```
Comment (ChakshuGautam) [inline packages/common-lib/src/components/FilterButton.js:59]: Why is the `filterButtonText` not translated here?

File: packages/common-lib/src/components/FilterButton.js [hunk 3]
```diff
@@ -78,7 +95,12 @@ export default function FilterButton({
                         )
                       }
                     }}
-                    {..._button}
+                    {...(isSelect > 0
+                      ? { ..._button, bg: 'button.500' }
+                      : _button)}
+                    {...(isSelect > 0
+                      ? { ..._optionButton, bg: 'button.500' }
+                      : _optionButton)}
```
Comment (ChakshuGautam) [inline packages/common-lib/src/components/FilterButton.js:103]: Simplify this by creating a new function. This is difficult to read.

File: packages/common-lib/src/components/FilterButton.js [hunk 4]
```diff
@@ -102,20 +124,29 @@ export default function FilterButton({
                 colorScheme='button'
                 variant='outline'
                 px='5'
+                _text={{
+                  textTransform: 'inherit',
+                  fontWeight: '500',
+                  fontSize: '14px'
+                }}
```
Comment (ChakshuGautam) [inline packages/common-lib/src/components/FilterButton.js:131]: Possible to use the same H1, H2 that we created earlier here.

File: packages/common-lib/src/components/FilterButton.js [hunk 5]
```diff
@@ -102,20 +124,29 @@ export default function FilterButton({
...
+                  if (isResettableFilter) {
+                    setGroupValue({})
+                    if (getObject) getObject({})
+                  }
                 }}
+                {..._button}
+                {..._resetButton}
               >
-                {t('RESET_FILTER')}
+                {resetButtonText ? resetButtonText : t('RESET_FILTER')}
```
Comment (ChakshuGautam) [inline packages/common-lib/src/components/FilterButton.js:149]: Why is the `resetButtonText` not translated here?

File: packages/common-lib/src/components/FilterButton.js [hunk 6]
```diff
@@ -130,7 +161,7 @@ export default function FilterButton({
           <HStack justifyContent={'space-between'}>
             <Stack p={5} pt={2} pb='25px'>
               <Text fontSize='16px' fontWeight={'600'}>
-                {t('SELECT_MODULE')}
+                {`${t('SELECT')} ${filterData?.name ? filterData?.name : ''}`}
```
Comment (ChakshuGautam) [inline packages/common-lib/src/components/FilterButton.js:164]: Why no translation for default?

File: packages/common-lib/src/components/Tab.js [hunk 7]
```diff
@@ -10,19 +14,26 @@ export default function Tab({ routes, _box }) {
...
-                p='3'
-                cursor='pointer'
-              >
+              <Box alignItems='center' p='3' cursor='pointer'>
                 <Animated.Text>
-                  <Text {...{ color: index === i ? 'button.500' : '#a1a1aa' }}>
+                  <Text
+                    {...{ color: index === i ? 'button.500' : 'blueGray.400' }}
+                    fontWeight='500'
+                  >
```
Comment (ChakshuGautam) [inline packages/common-lib/src/components/Tab.js:22]: Use existing text styles that we build or create new.

File: packages/common-lib/src/components/theme.ts [hunk 8]
```diff
@@ -94,7 +94,8 @@ const DEFAULT_THEME = {
       500: '#FDDFD8',
       600: '#feefeb',
       700: '#B9FBC0',
-      800: '#FDF2EE'
+      800: '#FDF2EE',
```
Comment (ChakshuGautam) [inline packages/common-lib/src/components/theme.ts:97]: I think this is updated in a separate PR. Can you link the PR.

File: packages/teacher-app/src/App.js [hunk 9]
```diff
@@ -45,23 +49,31 @@ function App() {
 
   const routes = [
```
Comment (ChakshuGautam) [inline packages/teacher-app/src/App.js:50]: Let's move routes to a new file called `routes.js` they have become big enough.

File: packages/worksheet/src/App.js [hunk 10]
```diff
@@ -16,21 +18,29 @@ function App() {
   );
   const routes = [
```
Comment (ChakshuGautam) [inline packages/worksheet/src/App.js:19]: Move `routes` to a new file.

File: packages/worksheet/src/components/QuestionBox/index.js [hunk 11]
```diff
@@ -59,14 +74,18 @@ const QuestionBox = ({ questionObject, selectData, setSelectData, _box }) => {
                     fontSize="14"
                     fontWeight="400"
                     textTransform="inherit"
-                    color={item.answer ? "successAlertText.500" : ""}
+                    color={
+                      item.answer && !isAnswerHide ? "successAlertText.500" : ""
+                    }
                   >
                     {alphabet[index] + ". "}
                   </Text>
```
Comment (ChakshuGautam) [inline packages/worksheet/src/components/QuestionBox/index.js:82]: Change text to prebuilt texts.

File: packages/worksheet/src/components/RecipientList.js [hunk 12]
```diff
@@ -0,0 +1,237 @@
...
+      </Box>
+    </Stack>
+  );
+}
+
+export const StudentList = ({ setPageName, students, setStudents }) => {
+  const { t } = useTranslation();
+  const Card = React.lazy(() => import("students/Card"));
+
+  const newStudents = [
```
Comment (ChakshuGautam) [inline packages/worksheet/src/components/RecipientList.js:121]: What is this?

File: packages/worksheet/src/components/WorksheetBox/index.js [hunk 13]
```diff
@@ -0,0 +1,222 @@
...
+                <HStack space={1} alignItems="center">
+                  <IconByName
+                    name="Heart3FillIcon"
+                    color="red.500"
+                    _icon={{ size: 12 }}
+                    isDisabled
+                  />
+                  <Text fontWeight="600" fontSize="10px">
+                    {item.likes + " likes"}
+                  </Text>
```
Comment (ChakshuGautam) [inline packages/worksheet/src/components/WorksheetBox/index.js:63]: Use text types.

File: packages/worksheet/src/config/worksheet.js [hunk 14]
```diff
@@ -19,7 +19,61 @@ export const worksheets = [
...
+    subject: "Math",
+    level: "Beginner",
+    grade: "VI",
+    questions: "20",
+    chapter: "2",
+    downloads: "2",
+  },
+];
+
+export const worksheetsList = [
```
Comment (ChakshuGautam) [inline packages/worksheet/src/config/worksheet.js:38]: Why is this list added?

File: packages/worksheet/src/pages/CreateWorksheet.js [hunk 15]
```diff
@@ -16,25 +16,87 @@ import {
...
   FormControl,
   Input,
+  ScrollView,
 } from "native-base";
 import React from "react";
 import { useTranslation } from "react-i18next";
 import { getAllQuestions } from "services";
 import manifest from "../manifest.json";
 
+const defaultInputs = [
```
Comment (ChakshuGautam) [inline packages/worksheet/src/pages/CreateWorksheet.js:29]: Move this to a new file. Refactor this into a smaller file. No more than 150 lines.

Comment (ChakshuGautam) [reply packages/worksheet/src/pages/CreateWorksheet.js:29]: Refactor all large files to smaller files.

File: packages/common-lib/src/components/FilterButton.js (hunk 1 above)
Comment (sagarkoshti1990) [reply packages/common-lib/src/components/FilterButton.js:49]: that is in another PR

File: packages/common-lib/src/components/FilterButton.js (hunk 2 above)
Comment (sagarkoshti1990) [reply packages/common-lib/src/components/FilterButton.js:59]: we can directly send the translated data as prop like `filterButtonText={t("FILTER_BUTTON_TEXT")}`

File: packages/common-lib/src/components/FilterButton.js (hunk 4 above)
Comment (sagarkoshti1990) [reply packages/common-lib/src/components/FilterButton.js:131]: that is in another PR

File: packages/worksheet/src/components/WorksheetBox/index.js (hunk 13 above)
Comment (sagarkoshti1990) [reply packages/worksheet/src/components/WorksheetBox/index.js:63]: how?

File: packages/worksheet/src/config/worksheet.js (hunk 14 above)
Comment (sagarkoshti1990) [reply packages/worksheet/src/config/worksheet.js:38]: its static data for list of worksheet now its not required so i will delete this.

File: packages/worksheet/src/pages/CreateWorksheet.js (hunk 15 above)
Comment (sagarkoshti1990) [reply packages/worksheet/src/pages/CreateWorksheet.js:29]: I am already made this in worksheet create flow PR

File: packages/worksheet/src/components/WorksheetBox/index.js (hunk 13 above)
Comment (sagarkoshti1990) [reply packages/worksheet/src/components/WorksheetBox/index.js:63]: that is in another PR https://github.com/shiksha-platform/frontend-modulefederation/pull/123

File: packages/common-lib/src/components/FilterButton.js [hunk 16]
```diff
@@ -15,17 +15,28 @@ import IconByName from './IconByName'
...
   const [filterData, setFilterData] = React.useState(false)
   const [selectData, setSelectData] = React.useState([])
 
+  React.useState(() => {
+    if (object) setGroupValue(object)
+  }, [object])
+
   return (
     <Box bg='white' roundedBottom={'xl'} {..._box}>
       <HStack justifyContent='end' alignItems='center'>
```
[the remark this answers is not in the document]
Comment (sagarkoshti1990) [reply packages/common-lib/src/components/FilterButton.js:42]: Not understand.

File: packages/worksheet/src/components/QuestionBox/index.js (hunk 11 above)
Comment (sagarkoshti1990) [reply packages/worksheet/src/components/QuestionBox/index.js:82]: Not Understand

**Comment by Shruti3004** [comment]
hey @sagarkoshti1990 can you resolve the conflictsby rebasing the PR?

**Pull request merged**